File: crawler/src/entity/log.py
# 百工谱 — Log ORM 实体
# 映射 crawler 库的 logs 表（与 crawler/init.sql 列一一对应）

from datetime import datetime

from sqlalchemy import BigInteger, DateTime, String, Text
from sqlalchemy.orm import Mapped, mapped_column

from src.entity.base import Base


class Log(Base):
    """logs 表 ORM 实体"""

    __tablename__ = "logs"

    id: Mapped[int] = mapped_column(BigInteger, primary_key=True)
    trace_id: Mapped[int | None] = mapped_column(BigInteger)
    user_id: Mapped[int] = mapped_column(BigInteger, nullable=False)
    user_name: Mapped[str] = mapped_column(String(8), nullable=False)
    user_ip: Mapped[str | None] = mapped_column(String(64))
    # level / request_method 是 PostgreSQL enum，用字符串存枚举名
    level: Mapped[str] = mapped_column(String(32), nullable=False)
    request_method: Mapped[str | None] = mapped_column(String(32))
    request_url: Mapped[str | None] = mapped_column(String(256))
    error_msg: Mapped[str | None] = mapped_column(Text)
    detail: Mapped[str | None] = mapped_column(Text)
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), nullable=False)
    updated_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), nullable=False)
    deleted_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True))

File: occupation/src/main/java/com/baigon/occupation/entity/Log.java
// 百工谱 — occupation 业务日志实体
package com.baigon.occupation.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.Table;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.type.SqlTypes;

@Entity
@Table(name = "logs")
public class Log extends BaseEntity {

    @Column(name = "trace_id")
    private Long traceId;
    @Column(name = "user_id", nullable = false)
    private Long userId;
    @Column(name = "user_name", nullable = false)
    private String userName;
    @Column(name = "user_ip")
    private String userIp;

    @Enumerated(EnumType.STRING)
    @JdbcTypeCode(SqlTypes.NAMED_ENUM)
    @Column(name = "level")
    private Level level;

    @Enumerated(EnumType.STRING)
    @JdbcTypeCode(SqlTypes.NAMED_ENUM)
    @Column(name = "request_method")
    private RequestMethod requestMethod;

    @Column(name = "request_url")
    private String requestUrl;
    @Column(name = "error_msg")
    private String errorMsg;
    @Column(name = "detail")
    private String detail;
    public enum Level { INFO, ERROR, WARNING }
    public enum RequestMethod { GET, POST, PUT, DELETE }

    public Long getTraceId() { return traceId; }
    public void setTraceId(Long traceId) { this.traceId = traceId; }
    public Long getUserId() { return userId; }
    public void setUserId(Long userId) { this.userId = userId; }
    public String getUserName() { return userName; }
    public void setUserName(String userName) { this.userName = userName; }
    public String getUserIp() { return userIp; }
    public void setUserIp(String userIp) { this.userIp = userIp; }
    public Level getLevel() { return level; }
    public void setLevel(Level level) { this.level = level; }
    public RequestMethod getRequestMethod() { return requestMethod; }
    public void setRequestMethod(RequestMethod requestMethod) { this.requestMethod = requestMethod; }
    public String getRequestUrl() { return requestUrl; }
    public void setRequestUrl(String requestUrl) { this.requestUrl = requestUrl; }
    public String getErrorMsg() { return errorMsg; }
    public void setErrorMsg(String errorMsg) { this.errorMsg = errorMsg; }
    public String getDetail() { return detail; }
    public void setDetail(String detail) { this.detail = detail; }
}
